lib/internal/http2/core.js

```diff
@@ -2145,7 +2190,7 @@ function clientSessionOnError(error) {
     return;
   this.destroy();
   this.removeListener('error', clientSocketOnError);
-  this.emit('error', error);
+  this.removeListener('error', clientSessionOnError);
```

Comment:
because of the `this.destroy()` above, that should be fine. Nothing that we need to catch should throw after this point.